x-pack/plugins/apm/server/lib/index_pattern/index.ts

```diff
@@ -5,21 +5,24 @@
  */
 import { InternalCoreSetup } from 'src/core/server';
 import { getSavedObjectsClient } from '../helpers/saved_objects_client';
-import indexPattern from '../../../../../../src/legacy/core_plugins/kibana/server/tutorials/apm/index_pattern.json';
+import apmIndexPattern from '../../../../../../src/legacy/core_plugins/kibana/server/tutorials/apm/index_pattern.json';
 
-export async function ensureIndexPatternExists(core: InternalCoreSetup) {
+export async function getIndexPattern(core: InternalCoreSetup) {
   const { server } = core.http;
   const config = server.config();
   const apmIndexPatternTitle = config.get('apm_oss.indexPattern');
   const savedObjectsClient = getSavedObjectsClient(server);
-  const savedObjects = [
-    {
-      ...indexPattern,
-      attributes: {
-        ...indexPattern.attributes,
+  try {
+    return await savedObjectsClient.get('index-pattern', apmIndexPattern.id);
+  } catch (error) {
+    // if GET fails, then create a new index pattern saved object
+    return await savedObjectsClient.create(
```

Comment:
Does `savedObjectsClient` return an error with a status code, e.g. 404? If so, then maybe we should check this before trying to create the index pattern. If it's some other kind of status code it seems like we'd probably want to surface it in the UI as an initialization failure or something.

Reply:
We attempt to lazily create the index pattern here as a convenience, but if a failure occurs, the APM UI shows the user some messaging that the APM index pattern is required to use certain features. The user is then directed to a tutorial where they can explicitly create the index pattern.